src/textioc/textioc.asm

```diff
@@ -0,0 +1,3975 @@
+;-------------------------------------------------------------
+; Includes
+;-------------------------------------------------------------
+	include '../include/library.inc'
+	include '../include/include_library.inc'
+
+;-------------------------------------------------------------
+	library 'TEXTIOC',2
+;-------------------------------------------------------------
+
+;-------------------------------------------------------------
+; Dependencies
+;-------------------------------------------------------------
+	include_library '../graphx/graphx.asm'
```

Comment:
Why is there still a dependency on graphx?

Reply:
I use gfx_FillRectangle_NoClip to draw the cursor in textio_Input

Reply:
What if I want to use this library on the homescreen to get text input?

Reply:
I remember looking at the homescreen cursor functions to see if TextIOC could support it, and I believe it lacked some of the necessary functions. I might be misremembering, however.

Reply:
You shouldn't have to do much for homescreen support. Just provide wrappers to the functions that you need; and abstract them. Don't bother writing an example or trying to implement a homescreen example, that's for the user to do. Keep in mind since you are using GetCSC, input is already horridly slow, so a few clocks makes zero difference.

Reply:
Although I would like to make *TextIOC* as extensible as possible, implementing homescreen routines would require resolving a great deal of compatibility issues, including translating pixel coordinates to row-column coordinates. The amount of time I have to do calculator programming is also going to diminish substantially for an indefinite period in the next few days, so I would like to complete this version although I understand that if I do implement them in a version 3, there might be some chaos regarding new programs using outdated *TextIOC* versions. If you would like, you are more than welcome to add homescreen support to *TextIOC* either in this version or a new one.

Reply:
All right fair enough :)

Reply:
Keep it in your todolist for the next version, also writing somewhere (readme?) as a caveat that homescreen stuff isn't supported yet :P

Anyway, is the current state ready to merge?

Reply:
I will do that. :) I will look over the examples and documentation one last time to check for any typos or any documentation that did not get updated during the development process.